src/server.cpp

```diff
@@ -3921,6 +3915,8 @@ std::string Server::getBuiltinLuaPath()
 
 void Server::setAsyncFatalError(const std::string &error)
 {
+	// print error right here in the thread that set it, for clearer logging
+	infostream << error << std::endl;
```

Comment:
This line can be difficult to track down when coming from log files. Please add a prefix that can be `grep`ped for.